consensus/src/main/java/org/apache/iotdb/consensus/ratis/ApplicationStateMachineProxy.java

```diff
@@ -142,13 +144,22 @@ public long takeSnapshot() throws IOException {
     // require the application statemachine to take the latest snapshot
     String metadata = Utils.getMetadataFromTermIndex(lastApplied);
     File snapshotDir = snapshotStorage.getSnapshotDir(metadata);
+
+    // delete snapshotDir fully in case of last failed take snapshot
+    FileUtils.deleteFully(snapshotDir);
+
     snapshotDir.mkdir();
     if (!snapshotDir.isDirectory()) {
       logger.error("Unable to create snapshotDir at {}", snapshotDir);
       return RaftLog.INVALID_LOG_INDEX;
     }
-    boolean success = applicationStateMachine.takeSnapshot(snapshotDir);
-    if (!success) {
+
+    boolean applicationTakeSnapshotSuccess = applicationStateMachine.takeSnapshot(snapshotDir);
+    boolean addTermIndexMetafileSuccess =
+        snapshotStorage.addTermIndexMetaFile(snapshotDir, metadata);
```

Comment:
Do we need to clean up this if failed?

Reply:
If takeSnapshot() failed, the whole `snapshotDir` will be deleted, including this TermIndexMetaFile.